3-networks/modules/vpn-ha/README.md

```diff
@@ -0,0 +1,100 @@
+# High Availability VPN module
+
+This module implementes the recomendation proposed in
+[High Availability VPN](https://cloud.google.com/network-connectivity/docs/vpn/concepts/topologies#overview).
+
+## Prerequisites
+
+1. Execution of at least one of the environments in `3-networks/env/`.
+1. Creation on two Cloud Routers for each Region. Both the `Restricted shared VPC`
+and the `Standard shared VPC` modules create two Cloud Routers for each Region
+that should be used as input for this module. **This module only works with two regions.**
+
+## Usage
+
+For each environmnet and for each network, you should add the following peace of code in the
+`envs/<environment>/main.tf`
+
+```hcl
+module "shared_<private/restricted>_vpn" {
+  source = "../../modules/vpn-ha"
+
+  project_id = local.<private/restricted>_project_id
+  default_region1 = var.default_region1
+  default_region2 = var.default_region2
+  vpc_label = "<private/restricted>"
+  environment_code = ${local.environment_code}
+
+  bgp_peer_secret = "<my-secret-value>"
+
+  on_prem_router_ip_address1 = "8.8.8.8" # on-prem router ip address
+  on_prem_router_ip_address2 = "8.8.8.8" # on-prem router ip address
+
+  bgp_peer_asn = "64515"
+
+  # tunnel 0
+  bgp_peer_address0 = "169.254.1.1"
+  bgp_peer_range0 = "169.254.1.2/30"
+
+  # tunnel 1
+  bgp_peer_address1 = "169.254.2.1"
+  bgp_peer_range1 = "169.254.2.2/30"
+
+  # tunnel 2
+  bgp_peer_address2 = "169.254.4.1"
+  bgp_peer_range2 = "169.254.4.2/30"
+
+  # tunnel 3
+  bgp_peer_address3 = "169.254.6.1"
+  bgp_peer_range3 = "169.254.6.2/30"
+
+  # tunnel 4
+  bgp_peer_address4 = "169.254.8.1"
```

Comment:
I think it would be better for this to reference the region, otherwise its unclear reading the code which region / tunnel it belongs too

Reply:
Agree!